lib/Logger/LogThread.cpp

```diff
@@ -50,9 +56,11 @@ bool LogThread::log(LogGroup& group, std::unique_ptr<LogMessage>& message) {
       (message->_level == LogLevel::FATAL || message->_level == LogLevel::ERR ||
        message->_level == LogLevel::WARN);
 
-  if (!_messages.push({&group, message.get()})) {
+  if (_pendingMessages.load() >= MAX_QUEUED_MESSAGES ||
+      !_messages.push({&group, message.get()})) {
     return false;
   }
+  _pendingMessages.fetch_add(1);
```

Comment:
Suggestion:
instead of performing an atomic load followed by an atomic fetch_add in 99.9999% of cases, we should unconditionally execute the fetch_add, then do the comparison, and roll back the fetch_add in case we are over the limit or something fails.
This will be more efficient.
```suggestion
  auto numMessages = _pendingMessages.fetch_add(1, std::memory_order_relaxed) + 1;
  if (numMessages >= MAX_QUEUED_MESSAGES ||
      !_messages.push({&group, message.get()})) {
      /* roll back the update */
    _pendingMessages.fetch_sub(1, std::memory_order_relaxed);
    return false;
  }
```